data/templates/rsyslog/rsyslog.conf.j2

```diff
@@ -1,78 +1,123 @@
+module(load="imklog")    # provides kernel logging support
 
-# We always log to /var/log/messages
-$outchannel global,/var/log/messages,262144,/usr/sbin/logrotate {{ logrotate }}
-{% if global.facility is vyos_defined %}
-{%     set tmp = [] %}
-{%     for facility, facility_options in global.facility.items() %}
-{%         set _ = tmp.append(facility.replace('all', '*') + '.' + facility_options.level.replace('all', '*')) %}
-{%     endfor %}
-{{ tmp | join(';') }} :omfile:$global
+# Import logs from journald
+module(
+    load="imjournal"
+    StateFile="/var/spool/rsyslog/imjournal.state"  # Persistent state file to track the journal cursor
+    Ratelimit.Interval="0"  # Disable rate limiting (set to "0" for no limit)
+    RateLimit.Burst="0"
+)
+
+###########################
+#### GLOBAL DIRECTIVES ####
+###########################
+# Log specific programs to auth.log, then stop further processing
+if (
+    $programname == "CRON" or
+    $programname == "sudo" or
+    $programname == "su"
+) then {
+    action(type="omfile" file="/var/log/auth.log")
+    stop
+}
+
+global(workDirectory="/var/spool/rsyslog")
+
+###############
+#### RULES ####
+###############
+
+# Send emergency messages to all logged-in users
+*.emerg action(type="omusrmsg" users="*")
+
+{% if marker is vyos_defined %}
+# Load the immark module for periodic --MARK-- message capability
+module(load="immark" interval="{{ marker.interval }}")
+{% endif %}
+{% if preserve_fqdn is vyos_defined %}
+# Preserve the fully qualified domain name (FQDN) in log messages
+global(preserveFQDN="on")
+{%     if preserve_fqdn.host_name is vyos_defined and preserve_fqdn.domain_name is vyos_defined %}
+# Set the local hostname for log messages
+global(localHostname="{{ preserve_fqdn.host_name }}.{{ preserve_fqdn.domain_name }}")
+{%     endif %}
 {% endif %}
 
-{% if file is vyos_defined %}
-# File based configuration section
-{%     for file_name, file_options in file.items() %}
-{%         set tmp = [] %}
-$outchannel {{ file_name }},/var/log/user/{{ file_name }},{{ file_options.archive.size }},/usr/sbin/logrotate {{ logrotate }}
-{%         if file_options.facility is vyos_defined %}
-{%             for facility, facility_options in file_options.facility.items() %}
-{%                 set _ = tmp.append(facility.replace('all', '*') + '.' + facility_options.level.replace('all', '*')) %}
-{%             endfor %}
-{%         endif %}
-{{ tmp | join(';') }} :omfile:${{ file }}
-{%     endfor %}
+#### GLOBAL LOGGING ####
+{% if local.facility is vyos_defined %}
+{%     set tmp = [] %}
+{%     if local.facility is vyos_defined %}
+{%         for facility, facility_options in local.facility.items() %}
+{%             set _ = tmp.append(facility.replace('all', '*') ~ "." ~ facility_options.level.replace('all', 'debug')) %}
+{%         endfor %}
+if prifilt("{{ tmp | join(',') }}") then {
+    action(
+        type="omfile"
+        file="/var/log/messages"
+        queue.size="262144"
```

Comment:
Can we include some comment about this number? If I see a non-round number, I always start wondering if it's either completely random (and can be arbitrary) or it was calculated in some complex and precise way (and dangerous to change).

Reply:
It's a relict from our history https://github.com/vyos/vyos-1x/blob/crux/src/conf_mode/syslog.py#L132

Reply:
Adjusted